pkg/services/ngalert/remote/remote_primary_forked_alertmanager.go

```diff
@@ -76,10 +76,20 @@ func (fam *RemotePrimaryForkedAlertmanager) TestTemplate(ctx context.Context, c
 	return fam.remote.TestTemplate(ctx, c)
 }
 
-func (fam *RemotePrimaryForkedAlertmanager) CleanUp() {}
+func (fam *RemotePrimaryForkedAlertmanager) CleanUp() {
+	// No cleanup to do in the remote Alertmanager.
+	fam.internal.CleanUp()
+}
 
-func (fam *RemotePrimaryForkedAlertmanager) StopAndWait() {}
+func (fam *RemotePrimaryForkedAlertmanager) StopAndWait() {
+	fam.internal.StopAndWait()
```

Comment:
How does this work? Do these calls block? Is there no error that can be returned if it fails?

Reply:
`StopAndWait()` stops the inhibitor, dispatcher, closes a few channels and waits using a waitgroup. It's blocking during that `waitgroup.Wait()` call, but this call to `internal.StopAndWait()` is no different from [the one we currently have](https://github.com/grafana/grafana/blob/68e37c3925080cf64a5e7570eabb042f19ca2dbf/pkg/services/ngalert/notifier/alertmanager.go#L158-L160). No values are returned.

Reply:
Right, but now we are communicating over a network when we do `fam.remote.StopAndWait()`. Do we need some kind of timeout via ctx?

Reply:
No, we don't have anything to stop and wait on the remote Alertmanager side. The only thing we stop here are our senders (goroutines forwarding alerts to the remote Alertmanager).

Reply:
Ah I understand, thanks for clarification. I thought `fam.remote.StopAndWait()` did a network call.